### PR TITLE
`mssql_managed_instance` - upgrade to `2023-08-01-preview` and `hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by catriona-m**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR updates the  `mssql_managed_instance` to `2023-08-01-preview`. It also updates it to use `hashicorp/go-azure-sdk` which is using the new base layer. 

To workaround the property name change from `StorageAccountType` to `RequestedBackupStorageRedundancy` and the change in value names, this pr takes the old value names and maps them to the new names to avoid a breaking change.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)
Tests running in TC

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

`mssql_managed_instance` - upgrade to `2023-08-01-preview` and `hashicorp/go-azure-sdk` [GH-26872]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go [hunk 1]
```diff
@@ -653,3 +609,32 @@ func (r MsSqlManagedInstanceResource) normalizeSku(sku string) string {
...
+func storageAccTypeToBackupStorageRedundancy(storageAccountType string) managedinstances.BackupStorageRedundancy {
+	switch storageAccountType {
+	case string(sql.StorageAccountTypeZRS):
+		return managedinstances.BackupStorageRedundancyZone
+	case string(sql.StorageAccountTypeLRS):
+		return managedinstances.BackupStorageRedundancyLocal
+	case "GZRS":
+		return managedinstances.BackupStorageRedundancyGeoZone
+	}
+	return managedinstances.BackupStorageRedundancyGeo
```
Comment (jackofallops) [inline internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go:627]: Can we define local constants/enums for these? Seems a lot of overhead to import the `sql` package for them?

**Review by jackofallops — CHANGES_REQUESTED** [review]
Thanks @catriona-m - Just a few comments below (sorry if some are already outdated, I started the review before you pushed some changes)

**Review by jackofallops — APPROVED** [review]
Thanks for the changes @catriona-m - LGTM 👍

**Review by stephybun — APPROVED** [review]
Thanks @catriona-m and @jackofallops LGTM 🌶️

**Pull request merged**